From: Julian Andres Klode <julian.klode@canonical.com>
Date: Mon, 24 May 2021 11:45:45 +0200
Subject: Do not addend vendor dbx in MokListX(RT) mirroring

Our vendor dbx is 19KB; which is huge and causing a lot of
laptops to run out of EFI variable storage space, or MacBooks
will just hang, when mirroring to MokListXR

Our kernels currently do not read MokListXRT anyway, so not
mirroring the vendor dbx does not affect security of current
kernels. Newer kernels do support vendor dbx, but we can just
bake in the vendor dbx into the kernel images too, rather than
rely on an EFI variable to pass that info through to the kernel.

The kernel could even be told to read MokList{,X} directly, before
calling ExitBootServices, in which case we could stop mirroring
entirely.
---
 mok.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/mok.c b/mok.c
index 0ac3415..cd1fd48 100644
--- a/mok.c
+++ b/mok.c
@@ -118,8 +118,6 @@ struct mok_state_variable mok_state_variable_data[] = {
 		     EFI_VARIABLE_NON_VOLATILE,
 	 .no_attr = EFI_VARIABLE_RUNTIME_ACCESS,
 	 .categorize_addend = categorize_deauthorized,
-	 .addend = &vendor_deauthorized,
-	 .addend_size = &vendor_deauthorized_size,
 	 .flags = MOK_MIRROR_KEYDB |
 		  MOK_MIRROR_DELETE_FIRST |
 		  MOK_VARIABLE_LOG,
